public/app/plugins/panel/alertlist/UnifiedAlertList.tsx

```diff
@@ -60,14 +75,76 @@ export function UnifiedAlertList(props: PanelProps<UnifiedAlertListOptions>) {
     dashboard = getDashboardSrv().getCurrent();
   });
 
+  const stateList = useMemo(() => getStateList(props.options.stateFilter), [props.options.stateFilter]);
+  const { options, replaceVariables } = props;
+  const parsedOptions: UnifiedAlertListOptions = {
+    ...props.options,
+    alertName: replaceVariables(options.alertName),
+    alertInstanceLabelFilter: replaceVariables(options.alertInstanceLabelFilter),
+  };
+
+  const matcherList = useMemo(
+    () => parseMatchers(parsedOptions.alertInstanceLabelFilter),
+    [parsedOptions.alertInstanceLabelFilter]
+  );
+
+  useEffect(() => {
+    //when coming back from cusom group mode, we restore limits and filters
+    const comingBackFromCustom = previousMode === GroupMode.Custom && props.options.groupMode === GroupMode.Default;
```

Comment:
I'm not sure why this needs a `usePrevious()` – can we just dispatch `fetchAllPromAndRulerRulesAction` whenever we're on `default group` mode? What would happen if we don't use the `usePrevious()` hook?